packages/rlc-common/src/metadata/buildTsConfig.ts

```diff
@@ -4,39 +4,65 @@
 import { Project } from "ts-morph";
 import { RLCModel } from "../interfaces.js";
 
-const restLevelTsConfigInAzureSdkForJs: Record<string, any> = {
-  extends: "../../../tsconfig.package",
-  compilerOptions: {
-    outDir: "./dist-esm",
-    declarationDir: "./types"
-  },
-  include: ["src/**/*.ts"]
+const restLevelTsConfigInAzureSdkForJs: (
+  model: RLCModel
+) => Record<string, any> = function (model: RLCModel) {
+  if (model.options?.moduleKind === "esm") {
```

Comment:
These get confusing when having early returns and I think should be separated into different files to make it easier to reason about.

Reply:
I'll refactor this in a follow up effort, I went ahead and refactored package generation in this PR